### PR TITLE
First draft of Contribution Guidelines

### PR DESCRIPTION
**Pull request by spier**
With a growing community around GenRMQ, it could be worthwhile to invest into making it easier for people to contribute to this project and to engage with us.

As a first step I figured that writing Contribution Guidelines in a `CONTRIBUTING.md`.
I have not done this before, so I will take a lot of inspiration from other projects.

Anybody here got experience with writing such guidelines that wants to help?

### CODE REVIEW

**Comment by spier** [comment]
This PR is still working progress, so I am keeping it as a draft for now.

Just wanted to post it here already, so that we can talk about the idea and get feedback from everybody.

**Comment by vorce** [comment]
I like it. Short, to the point, and open to change if needed.

File: CONTRIBUTING.md
Comment (akoutmos) [inline CONTRIBUTING.md:15]: In situations like this, I personally prefer to omit the list of characteristics given that it is impossible to enumerate all the characteristics for which someone may be self-conscious. Perhaps a slightly modified version of the Golden Rule where we just pledge to treat everyone with respect?

```
We as members, contributors, and leaders pledge to make participation in our community a pleasant experience for everyone. All are welcome to participate and can expect to be treated with politeness and respect. Likewise, we expect participants to treat other members, contributors, leaders and participants with politeness and respect. Harassment for any reason based on immutable or mutable characteristics will not be tolerated.
```

**Comment by akoutmos** [comment]
Overall looks great! Thanks for adding this. Just 1 small comment :)

File: CONTRIBUTING.md
Comment (spier) [reply CONTRIBUTING.md:15]: Creating a complete list of such characteristics certainly seems hard. However I find them useful as examples. I took this introduction from the [Contributor Covenant v2.0](https://www.contributor-covenant.org/version/2/0/code_of_conduct/), assuming that they thought about this longer and harder than I could.

For the paragraph that you are quoting above, were did you get that from?

**Comment by spier** [comment]
Oops, I just realized that we already have a **code of conduct**.  So I should not duplicate this in `CONTRIBUTING.md`.

File: CONTRIBUTING.md
Comment (spier) [reply CONTRIBUTING.md]: @akoutmos I think we can actually park this conversation for now, as I realized only now that we already have a Code of Conduct in a separate file. 

So rather than changing the existing Code Of Conduct, I would like to focus this PR on introducing a first small version of our Contribution Guidelines.

**Comment by spier** [comment]
This is ready for your review now.

I removed the Code of Conduct from `CONTRIBUTING.md` as I realized that we already had one in a separate file (it is also using the Contributor Covenant but in an earlier version).

I find it is fairly important that we as the first group of maintainers/trusted committers are in agreement about how we want to shape this open source project. Therefore I want to give everybody enough time for review.

I am proposing a +1 voting on this PR within a timeframe of 1 week.
i.e. if nobody requests changes for this PR, and if at least 2 people (other than myself) sign it off, than I will consider it mergeable on Oct 24th (in 1 week from today).

**Comment by mkorszun** [comment]
Looks great. @spier just to give you a little bit of context: Code of Conduct comes from github.

**Review by mkorszun — APPROVED** [review]
Sounds good 👍

**Review by vorce — APPROVED** [review]

**Comment by spier** [comment]
> Looks great. @spier just to give you a little bit of context: Code of Conduct comes from github.

@mkorszun what do you mean by it "comes from github"?

**Comment by mkorszun** [comment]
@spier you can add Code of Conduct from here:
Screen Shot 2020-10-19 at 09 11 30
and this is what I have done

**Review by Shemeikka — APPROVED** [review]

**Comment by spier** [comment]
Due date for review of Oct 24th is today 👍 
The PR got 3 approvals and no strong objections, hence I am merging it.

Thank you all for the feedback!

**Comment by spier** [comment]
Success! 👍 

contribution-guildlines-activated

**Pull request merged**
